devito/data/decomposition.py

```diff
@@ -241,6 +249,25 @@ def convert_index(self, *args, rel=True):
                 if glb_idx_max is not None and glb_idx_max < 0:
                     glb_idx_max = self.glb_max + glb_idx_max + 1
                 # -> Do the actual conversion
+                # First check for the special case of a slice with negative step
+                if isinstance(glb_idx, slice):
+                    if glb_idx.step < 0:
```

Comment:
you can squash this ``if`` with the upper one?
```
if isinstance(glb_idx, slice) and glb_idx.step < 0:
    ...
```

Reply:
Cool, I just learned about 'short circuit evaluation'.